### PR TITLE
[DEV-1662] Homepage nextjs refactor

### PR DESCRIPTION
**Pull request by MikeAtUqido**
This PR depends on #909 

#### List of Changes

* Add the new ecosystem section

#### Motivation and Context

With the solutions the homepage needed to be refactored in order to display the new Ecosystem section

#### How Has This Been Tested?

Manually

#### Screenshots (if appropriate):
PagoPA-DevPortal-PagoPA-DevPortal
PagoPA-DevPortal-PagoPA-DevPortal (1)

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/_contents/translations.ts [hunk 1]
```diff
@@ -107,10 +108,52 @@ export const translations = {
...
     webinarBannerButtonContent: 'Scopri',
-    productsShowcase: {
+    ecosystem: {
       title: 'Scopri il nostro ecosistema',
-      products: [appIo, ioSign, pagoPa, send],
-    },
+      productsTabName: 'Per prodotti',
+      products: [
+        {
+          title: appIo.name,
```
Comment (tommaso1) [inline apps/nextjs-website/src/_contents/translations.ts:116]: ``Maybe it can be useful to use an util something like 
```
const productToEcosystemProduct = (product: Product) => ({
  title: product.name, text: product.description, href: product.slug, icon: product.logo.url
})
```
so that this array woul become something like 
```
products: [
  productToEcosystemProduct(appIo),
  productToEcosystemProduct(ioSign),
  productToEcosystemProduct(pagoPa),
  productToEcosystemProduct(send)
]
```

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx [hunk 2]
```diff
@@ -6,37 +6,61 @@ type TabPanelProps = {
   readonly children?: ReactNode;
   readonly index: number;
   readonly value: number;
+  readonly px?: number;
```
Comment (tommaso1) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:9]: maybe we could pass a style prop to have the possiblity to further modify this later?

File: apps/nextjs-website/src/lib/homepage.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { StrapiHomepage } from '@/lib/strapi/homepage';
 import { translations } from '@/_contents/translations';
 import { Webinar } from './types/webinar';
 import { webinars } from '@/_contents/webinars';
+import { CardsGridProps } from '@/components/molecules/CardsGrid/CardsGrid';
 
 export type HomepageProps = {
```
Comment (tommaso1) [inline apps/nextjs-website/src/lib/homepage.ts:8]: this type feels very heavy to read, maybe splitting it into subtypes can help?
```suggestion
type NewsShowcaseItemProps = {
  readonly comingSoon?: boolean;
  readonly title: string;
  readonly publishedAt?: Date;
  readonly link: {
    readonly text: string;
    readonly url: string;
    readonly target?: '_self' | '_blank' | '_parent' | '_top' | null;
  };
  readonly image: {
    readonly name: string;
    readonly alternativeText: string | null;
    readonly width: number;
    readonly height: number;
    readonly ext: string;
    readonly mime: string;
    readonly url: string;
  } | null;
};

type NewsShowcaseProps = {
  readonly title: string;
  readonly items: readonly NewsShowcaseItemProps[];
};

type CardsGridProps = {
  cards: CardsGridProps['cards'];
};

type EcosystemSolutionsCtaProps = {
  readonly variant?: 'text' | 'contained' | 'outlined';
  readonly link: {
    readonly href: string;
    readonly text: string;
    readonly target?: '_self' | '_blank' | '_parent' | '_top';
  };
};

type EcosystemProps = {
  readonly title: string;
  readonly productsTabName: string;
  readonly products: CardsGridProps['cards'];
  readonly solutionsTabName: string;
  readonly solutions: CardsGridProps['cards'];
  readonly solutionsCta?: EcosystemSolutionsCtaProps;
};

export type HomepageProps = {
  readonly hero: readonly CtaSlideProps[];
  readonly newsShowcase: NewsShowcaseProps;
  readonly ecosystem: EcosystemProps;
};
```

**Review by tommaso1 — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/_contents/translations.ts [hunk 4]
```diff
@@ -107,10 +108,52 @@ export const translations = {
...
+        },
+      ],
+      solutionsTabName: 'Per soluzioni',
+      solutions: [
+        {
+          title: send.name,
+          text: send.description,
+          href: send.slug,
+          icon: send.logo.url,
+        },
```
Comment (marcobottaro) [inline apps/nextjs-website/src/_contents/translations.ts:147]: This is not a solution, I think you should remove this object

File: apps/nextjs-website/src/_contents/translations.ts [hunk 5]
```diff
@@ -107,10 +108,52 @@ export const translations = {
...
+      ],
+      solutionsCta: {
+        variant: 'text',
+        link: {
+          text: 'Vai a soluzioni',
+          href: '/solutions',
+        },
+      },
+    } as HomepageProps['ecosystem'],
```
Comment (marcobottaro) [inline apps/nextjs-website/src/_contents/translations.ts:156]:
```suggestion
      solutionsTabName: 'Per soluzioni',
      solutions: [],
    },
```

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 6]
```diff
@@ -67,11 +68,17 @@ export const StrapiHomepageCodec = t.strict({
...
+        productsTabName: t.string,
         products: t.strict({
           data: t.array(ProductCodec),
         }),
+        solutionsTabName: t.string,
+        solutions: t.strict({
+          data: t.array(SolutionCodec),
+        }),
+        solutionsCta: t.union([NullToUndefinedCodec, CallToActionCodec]),
       }),
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:82]: I think you should manage the eventuality of a null value of the ecosystem object coming from Strapi
```suggestion
      ecosystem: t.union([
        NullToUndefinedCodec,
        t.strict({
          title: t.string,
          productsTabName: t.string,
          products: t.strict({
            data: t.array(ProductCodec),
          }),
          solutionsTabName: t.string,
          solutions: t.strict({
            data: t.array(SolutionCodec),
          }),
          solutionsCta: t.union([NullToUndefinedCodec, CallToActionCodec]),
        }),
      ]),
```

File: apps/nextjs-website/src/lib/homepage.ts [hunk 7]
```diff
@@ -86,16 +86,30 @@ export const makeHomepageProps = (
...
+        text: solution.attributes.description ?? '',
+        href: solution.attributes.slug,
+        icon: solution.attributes.icon.data.attributes.url,
+      })
+    ),
+    solutionsCta: strapiHomepage.data.attributes.ecosystem.solutionsCta && {
+      variant: strapiHomepage.data.attributes.ecosystem.solutionsCta.variant,
+      link: strapiHomepage.data.attributes.ecosystem.solutionsCta.link,
+    },
   },
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/homepage.ts:113]: As a consequence of the previous suggestion, you should manage the empty ecosystem here too:
```suggestion
  ...(strapiHomepage.data.attributes.ecosystem && {
    ecosystem: {
      title: strapiHomepage.data.attributes.ecosystem.title,
      productsTabName: strapiHomepage.data.attributes.ecosystem.productsTabName,
      products: strapiHomepage.data.attributes.ecosystem.products.data.map(
        (product) => ({
          title: product.attributes.name,
          text: product.attributes.description ?? '',
          href: product.attributes.slug,
          icon: product.attributes.logo.data.attributes.url,
        })
      ),
      solutionsTabName:
        strapiHomepage.data.attributes.ecosystem.solutionsTabName,
      solutions: strapiHomepage.data.attributes.ecosystem.solutions.data.map(
        (solution) => ({
          title: solution.attributes.title,
          text: solution.attributes.description ?? '',
          href: solution.attributes.slug,
          icon: solution.attributes.icon.data.attributes.url,
        })
      ),
      solutionsCta: strapiHomepage.data.attributes.ecosystem.solutionsCta && {
        variant: strapiHomepage.data.attributes.ecosystem.solutionsCta.variant,
        link: strapiHomepage.data.attributes.ecosystem.solutionsCta.link,
      },
    },
  }),
```

File: apps/nextjs-website/src/components/molecules/CardsGrid/CardsGrid.tsx [hunk 8]
```diff
@@ -29,12 +30,13 @@ const CardsGrid = ({
...
   cardSvg,
   cardSize,
+  containerPx,
 }: CardsGridProps) => {
   const { palette } = useTheme();
   const t = useTranslations('shared');
 
   return (
-    <EContainer>
+    <EContainer px={containerPx} containerSx={{ px: containerPx }}>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/molecules/CardsGrid/CardsGrid.tsx:39]: I don't see any difference with:
```suggestion
    <EContainer containerSx={{ px: containerPx }}>
```
why did you use both properties?

File: apps/nextjs-website/src/components/organisms/Ecosystem/Ecosystem.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+type EcosystemProps = {
+  title?: string;
+  backgroundColor?: string;
+  items: ReadonlyArray<TabItem>;
+};
+
+const Ecosystem = ({ title, backgroundColor, items }: EcosystemProps) => {
+  const theme = useTheme();
+  return (
+    <Box
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/organisms/Ecosystem/Ecosystem.tsx:17]: I think you should fix some distances:
image
image
image
image

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/_contents/translations.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (MikeAtUqido) [reply apps/nextjs-website/src/_contents/translations.ts:130]: It's necessary or else the static homepage raises errors

File: apps/nextjs-website/src/lib/strapi/codecs/SolutionCodec.ts
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/codecs/SolutionCodec.ts:1]: First of all i don't know if you have already see the Codec in [this PR](https://github.com/pagopa/developer-portal/pull/908/files#diff-8fa291cc37b964a997f61206f62dd48eb52859666a9bd694631c019d38b46c1c) but maybe is better to branch from that PR or will have to be carefull in future to manage eventual conflict.

Second of which this is great to have this Codec sperated in because will be used in other codec :+1:  But i think,  even work fine for SolutionDetailPage, it has way too many attribute for the necessity of homepage component i think is better to subdivide this Codec in more manageble chunks so which "Page" codec can use the best fitting codec  (this also simplify a lot populate logic).

I think that could be more helpful if it is subdicided like this:
```
export const SolutionCodec = t.strict({
  attributes: t.strict({
    slug: t.string,
    icon: t.strict({ data: MediaCodec }),
    kickerTitle: t.string,
    title: t.string,
    description: t.union([NullToUndefinedCodec, t.string]),
    steps: t.array(SolutionStepCodec),
    dirName: t.string,
    landingUseCaseFile: t.string,
    stats: t.array(SolutionStatCodec),
  }),
});

const WithRelationCodec = t.strict({
  attributes: t.strict({
    webinars: t.strict({ data: t.array(WebinarCodec) }),
    products: t.strict({ data: t.array(ProductCodec) }),
    bannerLinks: t.array(BannerLinkCodec),
  }),
});

export const SolutionWithRelationCodec = t.intersection([
  SolutionCodec,
  WithRelationCodec,
]);
```
In this way we can use `SolutionWithRelationCodec` for solution detail page codec and `SolutionCodec` for homepage codec

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 10]
```diff
@@ -102,6 +112,25 @@ const makeStrapiHomepagePopulate = () =>
...
+          'solutions.webinars',
+          'solutions.webinars.coverImage.image',
+          'solutions.webinars.relatedLinks.link',
+          'solutions.webinars.webinarSpeakers.avatar',
+          'solutions.bannerLinks',
+          'solutions.bannerLinks.icon',
+          'solutions.products',
+          'solutions.products.logo',
+          'solutionsCta.link',
+        ],
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:132]: I got an error when fetching data due to this line:
Screenshot from 2024-06-25 16-31-40
```suggestion
        populate: [
          'products.logo',
          'solutions.icon',
          'solutions.steps',
          'solutions.steps.products',
          'solutions.steps.products.logo',
          'solutions.stats',
          'solutions.webinars',
          'solutions.webinars.coverImage.image',
          'solutions.webinars.relatedLinks.links',
          'solutions.webinars.webinarSpeakers.avatar',
          'solutions.bannerLinks',
          'solutions.bannerLinks.icon',
          'solutions.products',
          'solutions.products.logo',
          'solutionsCta.link',
        ],
```

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/homepage.ts (hunk 10 above)
Comment (MarcoPonchia) [reply apps/nextjs-website/src/lib/strapi/homepage.ts:132]: The suggestion above should handle the problem, but i think that the issue is more complex is due to use a complete codec of solutions event when we need to fetch few data.  

NB: I suggest to resolve [this issue before](https://github.com/pagopa/developer-portal/pull/910/files#r1653011953)

I think is better only to fetch de need data and remove all unecessary includes:
```suggestion
        populate: ['products.logo', 'solutions.icon', 'solutionsCta.link'],
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by tommaso1 — APPROVED** [review]

File: apps/nextjs-website/src/lib/homepage.ts [hunk 11]
```diff
@@ -1,57 +1,67 @@
...
-    }[];
+  readonly image: {
+    readonly name: string;
+    readonly alternativeText: string | null;
+    readonly width: number;
+    readonly height: number;
+    readonly ext: string;
+    readonly mime: string;
+    readonly url: string;
+  } | null;
```
[the remark this answers is not in the document]
Comment (marcobottaro) [reply apps/nextjs-website/src/lib/homepage.ts:25]: Maybe you could do this refactor now, as the #908 has already been merged

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/__tests__/homepage.test.ts
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/homepage.test.ts:1]: I think you should move this file in a separate PR

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:1]: You should add a gray line under the title of unselected tabs like on Figma:
image

image

Comment (marcobottaro) [reply apps/nextjs-website/src/components/atoms/TabComponent/TabComponent.tsx:1]: You should also align properly products and solutions cards (cards should fit all width):

image

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 12]
```diff
@@ -9,6 +9,7 @@ import { MediaCodec } from './codecs/MediaCodec';
 import { LinkCodec } from './codecs/LinkCodec';
 import { WebinarCodec } from './webinars';
 import { ProductCodec } from './codecs/ProductCodec';
+import { SolutionCodec } from './codecs/SolutionCodec';
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:12]:
```suggestion
import { SolutionBaseAttributesCodec } from './codecs/SolutionCodec';
```

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 13]
```diff
@@ -67,12 +68,21 @@ export const StrapiHomepageCodec = t.strict({
...
         }),
       ]),
-      productsShowcase: t.strict({
-        title: t.string,
-        products: t.strict({
-          data: t.array(ProductCodec),
+      ecosystem: t.union([
+        NullToUndefinedCodec,
+        t.strict({
+          title: t.string,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:74]:
```suggestion
          title: t.union([NullToUndefinedCodec, t.string]),
```

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 14]
```diff
@@ -67,12 +68,21 @@ export const StrapiHomepageCodec = t.strict({
...
+          productsTabName: t.string,
+          products: t.strict({
+            data: t.array(ProductCodec),
+          }),
+          solutionsTabName: t.string,
+          solutions: t.strict({
+            data: t.array(SolutionCodec),
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:81]:
```suggestion
            data: t.array(SolutionBaseAttributesCodec),
```

File: apps/nextjs-website/src/lib/homepage.ts [hunk 15]
```diff
@@ -79,17 +89,34 @@ export const makeHomepageProps = (
...
-        name: product.attributes.name,
-        description: product.attributes.description,
-        slug: product.attributes.slug,
-        logo: product.attributes.logo.data.attributes,
-      })
-    ),
-  },
+  ...(strapiHomepage.data.attributes.ecosystem && {
+    ecosystem: {
+      title: strapiHomepage.data.attributes.ecosystem.title,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/homepage.ts:94]: The title can be undefined, so:
```suggestion
      title: strapiHomepage.data.attributes.ecosystem.title || '',
```

File: apps/nextjs-website/src/lib/homepage.ts [hunk 16]
```diff
@@ -79,17 +89,34 @@ export const makeHomepageProps = (
...
+          icon: product.attributes.logo.data.attributes.url,
+        })
+      ),
+      solutionsTabName:
+        strapiHomepage.data.attributes.ecosystem.solutionsTabName,
+      solutions: strapiHomepage.data.attributes.ecosystem.solutions.data.map(
+        (solution) => ({
+          title: solution.attributes.title,
+          text: solution.attributes.description ?? '',
+          href: solution.attributes.slug,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/homepage.ts:110]:
```suggestion
          href: `/solutions/${solution.attributes.slug}`,
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by marcobottaro — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Review by MarcoPonchia — APPROVED** [review]
